services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/taydennys/TaydennysServiceITest.kt

```diff
@@ -1026,6 +1031,77 @@ class TaydennysServiceITest(
+                hakemusFactory
+                    .builder(USERNAME, hankeEntity, ApplicationType.EXCAVATION_NOTIFICATION)
+                    .withMandatoryFields(hankealue = hanke.alueet.single())
+                    .withStatus(ApplicationStatus.WAITING_INFORMATION, alluId)
+                    .saveEntity()
+            val taydennys = taydennysFactory.builder(hakemus).withStartTime(startTime).save()
+            justRun { alluClient.respondToInformationRequest(any(), any(), any(), any()) }
+            every { alluClient.getApplicationInformation(any()) } returns
+                AlluFactory.createAlluApplicationResponse(
+                    hakemus.alluid!!,
+                    status = ApplicationStatus.HANDLING,
+                )
+            val attachments = mutableListOf<Attachment>()
+            every { alluClient.addAttachment(hakemus.alluid!!, capture(attachments)) } throws
+                AlluException()
+
+            taydennysService.sendTaydennys(taydennys.id, USERNAME)
+
+            val attachmentMetadata =
+                attachments
+                    .map { it.metadata }
+                    .first { it.name == "haitaton-haittojenhallintasuunnitelma-taydennys.pdf" }
+            assertThat(attachmentMetadata)
+                .prop(AttachmentMetadata::description)
+                .isNotNull()
+                .contains("Haittojenhallintasuunnitelma from Haitaton taydennys, dated ")
+            verifySequence {
+                alluClient.respondToInformationRequest(any(), any(), any(), any())
+                alluClient.addAttachment(hakemus.alluid!!, withName(FORM_DATA_PDF_FILENAME))
+                alluClient.addAttachment(hakemus.alluid!!, withName(HHS_PDF_FILENAME))
+                alluClient.getApplicationInformation(hakemus.alluid!!)
+            }
+        }
+
+        @Test
+        fun `updates the Allu status even when sending the haittojenhallintasuunnitelma attachment fails`() {
+            val hanke = hankeFactory.builder(USERNAME).withHankealue().save()
+            val hankeEntity = hankeRepository.getReferenceById(hanke.id)
+            val hakemus =
+                hakemusFactory
+                    .builder(USERNAME, hankeEntity, ApplicationType.EXCAVATION_NOTIFICATION)
+                    .withMandatoryFields(hankealue = hanke.alueet.single())
+                    .withStatus(ApplicationStatus.WAITING_INFORMATION, alluId)
+                    .saveEntity()
+            val taydennys = taydennysFactory.builder(hakemus).withStartTime(startTime).save()
+            justRun { alluClient.respondToInformationRequest(any(), any(), any(), any()) }
+            every { alluClient.getApplicationInformation(any()) } returns
+                AlluFactory.createAlluApplicationResponse(
+                    hakemus.alluid!!,
+                    status = ApplicationStatus.HANDLING,
+                )
+            justRun { alluClient.addAttachment(hakemus.alluid!!, any()) }
```

Comment:
Should this test throw Exception when sending the attachment to Allu?

Reply:
It looks like I changed the wrong test after copy-pasting. Fixed now. Thank you for noticing.